crates/bevy_text/src/text.rs

```diff
@@ -188,11 +180,107 @@ pub enum BreakLineOn {
     AnyCharacter,
 }
 
-impl From<BreakLineOn> for glyph_brush_layout::BuiltInLineBreaker {
-    fn from(val: BreakLineOn) -> Self {
-        match val {
-            BreakLineOn::WordBoundary => glyph_brush_layout::BuiltInLineBreaker::UnicodeLineBreaker,
-            BreakLineOn::AnyCharacter => glyph_brush_layout::BuiltInLineBreaker::AnyCharLineBreaker,
+#[derive(Clone, Debug, Reflect, FromReflect)]
+pub enum FontRef {
+    /// A reference to a font loaded as a bevy asset.
+    Asset(Handle<Font>),
+    /// A reference to a font queried by font family and attributes.
+    /// This is useful for example for fonts that are not loaded as a bevy asset,
+    /// such as system fonts.
+    // TODO: Support Reflect?
+    Query(#[reflect(ignore)] FontQuery),
+}
+
+impl From<Handle<Font>> for FontRef {
+    fn from(handle: Handle<Font>) -> Self {
+        Self::Asset(handle)
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct FontQuery {
```

Comment:
Definitely needs doc strings. I also think we should really try and avoid the `Query` name if possible, since this doesn't seem to use bevy_ecs::Query, which is going to be a source of a lot of confusion.

Reply:
Looking at this more, I think that "system font" is probably a better way to describe this functionality?

Reply:
`SystemFont` doesn't quite capture the idea as you can actually use a `FontQuery` to refer to a font you loaded as an asset (since it's always loaded into the `Database`), which I've confirmed works (but I probably need to test for edge cases). `FontQuery` better captures the idea, and it's [the terminology used by `fontdb`](https://docs.rs/fontdb/latest/fontdb/struct.Query.html), but `FontSearch` could maybe work.